cmd/kubeless/function.go

```diff
@@ -100,14 +102,107 @@ func parseMemory(mem string) (resource.Quantity, error) {
 	return quantity, nil
 }
 
-func getFunctionDescription(funcName, ns, handler, funcContent, deps, runtime, topic, schedule, runtimeImage, mem string, triggerHTTP bool, envs, labels []string, defaultFunction spec.Function) (f *spec.Function, err error) {
+func getFileSha256(file string) (string, error) {
+	var checksum string
+	h := sha256.New()
+	ff, err := os.Open(file)
+	if err != nil {
+		return checksum, err
+	}
+	defer ff.Close()
+	_, err = io.Copy(h, ff)
+	if err != nil {
+		return checksum, err
+	}
+	checksum = hex.EncodeToString(h.Sum(nil))
+	return checksum, err
+}
+
+// Convert MB to Bytes as int64
+func mbtoInt64(mb int) int64 {
+	return int64(mb * 1024 * 1024)
+}
+
+func uploadFunction(file string, cli kubernetes.Interface) (string, string, error) {
+	var checksum string
+	stats, err := os.Stat(file)
+	if stats.Size() > mbtoInt64(50) { // TODO: Make the max file size configurable
+		err = errors.New("The maximum size of a function is 50MB")
+		return "", "", err
+	}
+	checksum, err = getFileSha256(file)
+	if err != nil {
+		return "", "", err
+	}
+	url, err := minio.UploadFunction(file, checksum, cli)
+	if err != nil {
+		return "", "", err
+	}
+	return url, checksum, err
+}
+
+func isMinioAvailable(cli kubernetes.Interface) bool {
+	_, err := cli.Core().Services("kubeless").Get("minio", metav1.GetOptions{})
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func getFunctionDescription(funcName, ns, handler, file, deps, runtime, topic, schedule, runtimeImage, mem string, triggerHTTP bool, envs, labels []string, defaultFunction spec.Function, cli kubernetes.Interface) (*spec.Function, error) {
 
 	if handler == "" {
```

Comment:
just a suggestion: I would do it like this - first, copy the whole default function spec to the returned func spec; then eventually check if a parameter is not empty (handler, file, deps, etc) and modify the returned func spec accordingly

Reply:
I ended up not doing this to avoid the PR to grow even bigger, I will take that into account for next improvements